sound/soc/sof/compress.c

```diff
@@ -179,7 +180,12 @@ static int sof_compr_set_params(struct snd_soc_component *component,
 	if (!spcm)
 		return -EINVAL;
 
-	pcm = kzalloc(sizeof(*pcm), GFP_KERNEL);
+	data_size = sizeof(params->codec);
+
+	if (sizeof(*pcm) + data_size > SOF_IPC_MSG_MAX_SIZE)
```

Comment:
Better use `sdev->ipc->max_payload_size` instead `SOF_IPC_MSG_MAX_SIZE` to avoid surprises if this is used with other than IPC3?

Reply:
Ok, i will fix this.